### PR TITLE
Nodejs week2

### PR DESCRIPTION
**Pull request by ShravyaGurmitkal**

### CODE REVIEW

File: nodejs/week2/app.js [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    try {
+        const queryParam = req.query.q;
+        if(queryParam === undefined) {
+            res.send(documents)
+        } else {
+            const resultsArray = searchQuery(queryParam)
+            if (resultsArray.length !== 0) {
+                res.send(resultsArray);
+            } else {
+                res.status(400).send(`There are no items with searched value`);
```
Comment (itsgauravsaxena) [inline nodejs/week2/app.js:24]: 404 is more like 'not found'. 400 means bad request if that is what wanted to throw

**Review by itsgauravsaxena — COMMENTED** [review]
Looks good!!

**Review by itsgauravsaxena — APPROVED** [review]

File: nodejs/week2/app.js [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    }
+})
+
+//POST /search
+app.post('/search', (req,res) => {
+    try {
+       const requestFields = req.body.fields;
+       const queryParam = req.query.q;
+       let resultsArray = [];
+       if(requestFields !== undefined && queryParam !== undefined) {
```
Comment (RZhyvitskyi) [inline nodejs/week2/app.js:55]: Can we just write `if(requestFields && queryParam) {` ?

**Review by RZhyvitskyi — APPROVED** [review]

File: nodejs/week2/app.js (hunk 2 above)
Comment (ShravyaGurmitkal) [reply nodejs/week2/app.js:55]: Yeah, we can write like that, it will also work

File: nodejs/week2/app.js (hunk 1 above)
Comment (ShravyaGurmitkal) [reply nodejs/week2/app.js:24]: okay

**Pull request merged**